### PR TITLE
feat: search admin users

### PR DESCRIPTION
**Pull request by carissa-tang**
## Notion ticket link

[Search Admin Users](https://www.notion.so/uwblueprintexecs/Search-Admin-Users-ac5007e1a3604e6e987a36ed6da9e937)

## Implementation description
* Add search bar for admin table
   * Matches searches for name or email (case-insensitive)
* Fix layout of page  

## Steps to test
https://user-images.githubusercontent.com/69697180/203686351-7bfbef89-2de0-45b5-8564-9993b52cf980.mov

https://user-images.githubusercontent.com/69697180/203824647-97622346-515d-4ae3-959e-5a1c1e989d9c.mov

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/common/AdminUserTable.tsx [hunk 1]
```diff
@@ -42,25 +42,26 @@ const AdminUserTable = ({
...
-                >{`${user.firstName} ${user.lastName}`}</Text>
-              </Td>
-              <Td>
-                <Text noOfLines={1} style={{ display: "block" }}>
-                  {user.email}
-                </Text>
-              </Td>
-            </Tr>
-          ))}
+          {adminUsers &&
```
Comment (joyce-shi) [inline frontend/src/components/common/AdminUserTable.tsx]: is this necessary? doesn't it just not `map` if there's nothing to map?

File: frontend/src/components/common/icons.tsx [hunk 2]
```diff
@@ -187,10 +186,19 @@ export const AlertIcon = (): React.ReactElement => (
...
-    <circle cx="12" cy="16" r="1" />
+    <circle fill="currentColor" cx="12" cy="16" r="1" />
     <path
       fill="currentColor"
       d="M12 8a1 1 0 0 0-1 1v4a1 1 0 0 0 2 0V9a1 1 0 0 0-1-1z"
     />
   </Icon>
 );
+
+export const SearchBar = (): React.ReactElement => (
```
Comment (joyce-shi) [inline frontend/src/components/common/icons.tsx]: is this called the `SearchBar` icon in the original package? If not, can we name it consistently with the package name? Also Let's append Icon at the end of the name to be consistent with the other variables in this file.

File: frontend/src/components/common/Sidebar.tsx [hunk 3]
```diff
@@ -12,20 +12,14 @@ interface SidebarProps {
...
-      <VStack alignItems="left">
+      <VStack alignItems="left" w="100%">
         {pages.map((page, index) => (
           <SidebarItem key={index} page={page} />
         ))}
       </VStack>
-    </Box>
+    </VStack>
   );
 };
```
Comment (joyce-shi) [inline frontend/src/components/common/Sidebar.tsx:25]: very nice! ty for fixing :D

File: frontend/src/components/pages/AdminPage.tsx [hunk 4]
```diff
@@ -28,98 +33,117 @@ const pages: Page[] = [
...
-      <>
-        <LoadingState />
-      </>
-    );
-  if (error)
-    return (
-      <>
-        <ErrorState />
-      </>
+  const adminData = React.useMemo(() => {
```
Comment (joyce-shi) [inline frontend/src/components/pages/AdminPage.tsx]: nit: `adminData` -> `admins`?

File: frontend/src/components/pages/AdminPage.tsx [hunk 5]
```diff
@@ -28,98 +33,117 @@ const pages: Page[] = [
...
+          {data && !error && !loading && (
+            <Tabs marginTop={3}>
+              <TabList>
+                <Tab color={unselectedColor}>Admin</Tab>
+                <Tab color={unselectedColor}>Teachers</Tab>
+              </TabList>
+              <TabPanels>
+                <TabPanel>
+                  <AdminUserTable
+                    adminUsers={adminData?.map((user: AdminUser) =>
```
Comment (joyce-shi) [inline frontend/src/components/pages/AdminPage.tsx]: i know this part isn't your code, but is there a cleaner way to pass the value of queried admin users? without using the helper function?

**Review by joyce-shi — COMMENTED** [review]
Left some small comments!

File: frontend/src/components/common/AdminUserTable.tsx [hunk 6]
```diff
@@ -45,7 +45,7 @@ const AdminUserTable = ({
           {adminUsers.map((user, index) => (
             <Tr
               _hover={{ backgroundColor: "blue.300", color: "grey.50" }}
-              key={user.firstName}
+              key={user.email}
```
Comment (carissa-tang) [inline frontend/src/components/common/AdminUserTable.tsx:48]: unique key

**Review by joyce-shi — APPROVED** [review]
LGTM!

**Pull request merged**